base/deprecated.jl

```diff
@@ -1009,4 +1009,18 @@ export @vectorize_1arg, @vectorize_2arg
 @deprecate abs(M::SymTridiagonal) abs.(M)
 @deprecate abs(x::AbstractSparseVector) abs.(x)
 
+# #18218
+eval(Base.LinAlg, quote
+    function arithtype(T)
+        depwarn("arithtype is deprecated as it is no longer needed in Julia Base.",
```

Comment:
If packages are using this together with `promote_op` then the new way of doing things would be `promote_op(matprod, S, T)` instead of `promote_op(*, arithtype(S), arithtype(T))`. If packages are using it some other way, they would have to define their own replacement, I guess.